### PR TITLE
Strong-name Bicep.Types.Az

### PR DESCRIPTION
**Pull request by shenglol**
This is to unblock the Graph team.

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
     <RestoreLockedMode Condition="'$(CI)' == 'true'">true</RestoreLockedMode>
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
+    <EnlistmentRoot>$(MSBuildThisFileDirectory.TrimEnd('\\'))</EnlistmentRoot>
```
Comment (majastrz) [inline src/Directory.Build.props:12]: >EnlistmentRoot [](http://example.com/codeflow?start=5&length=14)

Is this needed?

**Review by majastrz — APPROVED** [review]
:shipit:

File: src/Bicep.Types.Az/Bicep.Types.Az.csproj [hunk 2]
```diff
@@ -8,6 +8,9 @@
     <Description>Bicep types generated from public Azure OpenAPI specs</Description>
   </PropertyGroup>
 
+    <!-- Need to sign the assembly with a strong name in Official Build. -->
+  <Import Project="$(EnlistmentRoot)\StrongName.props" />
```
[the remark this answers is not in the document]
Comment (shenglol) [reply src/Bicep.Types.Az/Bicep.Types.Az.csproj:12]: We have a condition in `StrongName.Props` so strong-name signing will only be done in official builds, so I think we should be fine?

File: src/Directory.Build.props (hunk 1 above)
Comment (shenglol) [reply src/Directory.Build.props:12]: Yeah it's used in `Bicep.Types.Az.csproj`. Copied this approach from one of our internal repos.

**Pull request merged**